### PR TITLE
Create separate plugin repository for Cloud Firestore plugins.

### PR DESCRIPTION
**Pull request by supriyapatil-google**
Initial copy of Firestore plugins from google-cloud repo.

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java [hunk 1]
```diff
@@ -383,7 +383,7 @@ private void validateDocumentLists(FailureCollector collector) {
     List<String> skipDocumentList = Splitter.on(',').trimResults().splitToList(getSkipDocuments());
 
     if (mode == SourceQueryMode.BASIC) {
-      if (!pullDocumentList.isEmpty() && !skipDocumentList.isEmpty()) {
+      if (pullDocumentList.isEmpty() && skipDocumentList.isEmpty()) {
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java:382]: Thanks for fixing this :)

File: pom.xml [hunk 2]
```diff
@@ -70,20 +70,11 @@
     <jee.version>7</jee.version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <avro.version>1.8.2</avro.version>
-    <bigquery.connector.hadoop2.version>hadoop2-1.0.0</bigquery.connector.hadoop2.version>
     <commons.codec.version>1.4</commons.codec.version>
     <cdap.version>6.3.0-SNAPSHOT</cdap.version>
```
Comment (CuriousVini) [inline pom.xml]: We can keep it 6.1.1

**Review by CuriousVini — APPROVED** [review]
Thanks for breaking it down in multiple commits! LGTM 👍 
Quick question - Is it tested on CDF environment?

File: pom.xml (hunk 2 above)
Comment (supriyapatil-google) [reply pom.xml]: Updated.

**Comment by supriyapatil-google** [comment]
Updated PR based on CDF testing. Please take a look.

File: src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java [hunk 3]
```diff
@@ -270,8 +259,12 @@ void validateFirestoreConnection(FailureCollector collector) {
         .withConfigProperty(FirestoreConstants.PROPERTY_COLLECTION)
         .withStacktrace(e.getStackTrace());
     } catch (Exception e) {
+      collector.addFailure(e.getMessage(), "Error while connecting to Firestore.")
```
Comment (CuriousVini) [inline src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java]: The second one is the action that should be taken, In this case we can prefix error with `Error while connecting to Firestore: {}, e.getMessage()` and keep action null

**Review by CuriousVini — APPROVED** [review]
Thanks, just one minor comments. Please fix and merge.

File: src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java (hunk 3 above)
Comment (supriyapatil-google) [reply src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java]: Done.

**Pull request merged**